src/apps/persistence/app_listing.clj

```diff
@@ -374,6 +384,25 @@
       ((partial query-spy "get-apps-for-user::search_query:"))
       select))
 
+;; Note: this function will be renamed when the performance improvement changes are complete.
+(defn new-get-apps-for-user
+  "Fetches Apps in all public groups and groups in `workspace`
+   (as returned by fetch-workspace-by-user-id),
+   marking whether each app is a favorite and including the user's rating in each app by the user_id
+   found in workspace.
+  If search_term is not empty, results are limited to apps that contain search_term in their name,
+   description, integrator_name, or tool name(s)."
+  [search_term {workspace_id :id :as workspace} favorites_group_index query_opts]
+  (let [app-ids (find-matching-app-ids search_term query_opts)]
+    (-> (get-base-app-listing-base-query workspace favorites_group_index query_opts)
```

Comment:
somewhat arbitrary placement for this comment:

One thing I've wanted to investigate, which either you already have or might find interesting, is if our use of the `app_listing` view is affecting performance in some fashion. Primarily, I worry that the DBMS might be doing some joins/scans/etc. both in the scan of the view and afterwards when things are joined on. That could possibly explain some of the benefits this PR is causing, since I see that the newly-implemented search query doesn't use the view (and these queries don't do additional joins outside of the view from what I can see); if so, it might provide a route that's even faster, if it allows re-combining the queries into one. Or I could be crazy and it makes no difference. Anyway!

Reply:
I experimented a bit with a query that was like the view with the additional joins included, and it made the performance worse, at least for the queries that I tried. Admittedly, I could have made a mistake, and I didn't experiment with it enough to determine why the performance was worse.

The primary reason that I'm doing the second query is because it allows me to join in the `tools` without breaking the grouping in the listing query. Adding the additional joins was significantly faster than doing the `exists subselect` to filter based on the tool name, but adding that in the original view would have broken the grouping for pipelines. Doing two queries allows me to remove the subselect without having to worry about the grouping at all.